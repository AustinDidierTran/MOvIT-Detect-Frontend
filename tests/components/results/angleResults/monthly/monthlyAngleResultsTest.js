/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import React from 'react';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import MonthlyAngleResults from '../../../../../src/components/results/angleResults/monthly/monthlyAngleResults';


Enzyme.configure({ adapter: new Adapter() });

describe('MonthlyAngleResults Tests', () => {
  const MONTH = '2';
  const initialState = { applicationReducer: { language: 'en' } };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    month: MONTH,
  };

  it('should have proptypes', () => {
    const actualValue = MonthlyAngleResults.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      month: PropTypes.string.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should update month', () => {
    const wrapper = shallow(<MonthlyAngleResults store={store} {...props} />).dive();
    wrapper.instance().componentWillReceiveProps({ month: '4' });

    expect(wrapper.state('month')).toEqual('4');
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<MonthlyAngleResults store={store} {...props} />).dive();

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
