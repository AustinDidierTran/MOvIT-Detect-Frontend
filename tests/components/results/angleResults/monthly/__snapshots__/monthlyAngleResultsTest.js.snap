// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyAngleResults Tests should match the snapshot 1`] = `
<div>
  <div
    className="col-lg-2 leftMenu"
  >
    <ul
      className="graphlist"
    >
      <li
        className="graphLink"
      >
        <a
          href="results/angle#monthlyAngle"
        >
          results.graphicsLink.angle.en
        </a>
      </li>
      <li
        className="graphLink"
      >
        <a
          href="results/angle#monthlyTilt"
        >
          SuccessfulTilt.tiltMade.en
        </a>
      </li>
    </ul>
  </div>
  <div
    className=" col-lg-10 offset-lg-2 results resultsContainer"
  >
    <div
      className="col-lg-8 offset-lg-1"
    >
      <div>
        <div>
          <Connect(MonthlyAngleDistribution)
            month="2"
          />
          <Connect(MonthlySuccessTilt)
            month="2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
